### PR TITLE
Add math formulation docs

### PR DESCRIPTION
**Pull request by charleskawczynski**
Add mathematical formulation documentation.

### CODE REVIEW

File: docs/src/Formulation.md [hunk 1]
```diff
@@ -0,0 +1,423 @@
+# Mathematical formulation
+
+The thermodynamics of moist air is often subject to empirical approximations, which usually are opaque, internally inconsistent, and/or inconsistent across model components. For example, microphysical process models often use different approximations for thermodynamic quantities such as saturation vapor pressures than the dynamical core. The often bewildering array of approximations makes it difficult to achieve global conservation, e.g., of energy, and it complicates the use of models for other planetary atmospheres, with different thermodynamic parameters.
+
+This chapter introduces one consistent set of thermodynamic approximations for all model components. The key to thermodynamic consistency at reasonable accuracy is to take the specific heat capacities of the constituents of moist air (dry air, water vapor, liquid water, and ice) to be constant, i.e., to assume the gases to be calorically perfect. We discuss how to derive all other thermodynamic quantities that are needed on the basis of this one approximation ([Romps2008](@cite),[Marquet2015](@cite)). This includes:
```
Comment (tapios) [inline docs/src/Formulation.md:5]:
```suggestion
Here we  introduce one consistent set of thermodynamic approximations for all model components. The key to thermodynamic consistency at reasonable accuracy is to take the specific heat capacities of the constituents of moist air (dry air, water vapor, liquid water, and ice) to be constant, i.e., to assume the gases to be calorically perfect. We discuss how to derive all other thermodynamic quantities that are needed on the basis of this one approximation ([Romps2008](@cite),[Marquet2015](@cite)). This includes:
```

File: docs/bibliography.bib [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+@article{Pressel2015,
+  title={Large-eddy simulation in an anelastic framework with closed water and entropy balances},
+  author={Pressel, Kyle G and Kaul, Colleen M and Schneider, Tapio and Tan, Zhihong and Mishra, Siddhartha},
+  journal={Journal of Advances in Modeling Earth Systems},
+  volume={7},
+  number={3},
+  pages={1425--1456},
+  year={2015},
+  publisher={Wiley Online Library}
```
Comment (tapios) [inline docs/bibliography.bib:50]:
```suggestion
```

File: docs/make.jl [hunk 3]
```diff
@@ -8,6 +11,8 @@ pages = Any[
     "Tested profiles" => "TestedProfiles.md",
     "Temperature profiles" => "TemperatureProfiles.md",
     "Developer docs" => "DevDocs.md",
+    "Mathematical formulation" => "Formulation.md",
```
Comment (tapios) [inline docs/make.jl:14]:
```suggestion
    "Thermodynamics overview" => "Formulation.md",
```

File: docs/src/Formulation.md [hunk 4]
```diff
@@ -0,0 +1,423 @@
...
+## Moist Static Energy
+
+The sum of the specific enthalpy of moist air and the specific gravitational potential energy ``Φ`` is the moist static energy [Neelin1987](@cite)
+```math
+\begin{equation}\label{e:MSE}
+\mathrm{MSE} = h + Φ.
+\end{equation}
+```
+
+Moist static energy arises naturally as the static energy component that is transported in moist air. ``Static`` here refers to the fact that the (small) kinetic energy contribution to the total energy is neglected (see section TODO). The global integral of moist static energy is approximately conserved in adiabatic processes, even in the presence of reversible phase transitions and latent heat release. It is also approximately materially conserved [Romps2015](@cite).
```
Comment (tapios) [inline docs/src/Formulation.md:197]:
```suggestion
Moist static energy arises naturally as the static energy component that is transported in moist air. ``Static`` here refers to the fact that the (small) kinetic energy contribution to the total energy is neglected (see section TODO). The global integral of moist static energy is approximately conserved in adiabatic processes, even in the presence of reversible phase transitions and latent heat release. It is also approximately materially conserved ([Romps2015](@cite)).
```

File: docs/src/Formulation.md [hunk 5]
```diff
@@ -0,0 +1,423 @@
...
+    p_v^* = p_{\mathrm{tr}} \left( \frac{T}{T_{\mathrm{tr}}} \right)^{\frac{\Delta c_p}{R_v}}
+        \exp \left[ \frac{L_0 - \Delta c_p T_0}{R_v}
+        \left( \frac{1}{T_{\mathrm{tr}}} - \frac{1}{T} \right) \right].
+        \label{e:SatVaporPressure}
+\end{equation}
+```
+
+With ``L_0 = L_{v,0}`` or ``L_0 = L_{s,0}`` and the corresponding heat capacity difference ``\Delta c_p``, this gives saturation vapor pressures over liquid or ice that are accurate within 3% for temperatures between 200K and 330K (with accuracy better than 1% for typical near-surface conditions).
+
+To obtain the saturation vapor pressure over a mixture of liquid and ice (e.g., in mixed-phase clouds), using a weighted average of the relevant specific latent heats in the vapor pressure \eqref{e:SatVaporPressure} leads to a thermodynamically consistent formulation [Pressel2015](@cite). That is, if a fraction ``λ_p`` of the condensate is liquid and the complement ``1-λ_p`` is ice, calculating the saturation vapor pressure with a specific latent heat ``λ_p L_v + (1-λ_p)L_s`` gives a thermodynamically consistent saturation vapor pressure over the mixture.
```
Comment (tapios) [inline docs/src/Formulation.md:221]:
```suggestion
To obtain the saturation vapor pressure over a mixture of liquid and ice (e.g., in mixed-phase clouds), using a weighted average of the relevant specific latent heats in the vapor pressure \eqref{e:SatVaporPressure} leads to a thermodynamically consistent formulation ([Pressel2015](@cite)). That is, if a fraction ``λ_p`` of the condensate is liquid and the complement ``1-λ_p`` is ice, calculating the saturation vapor pressure with a specific latent heat ``λ_p L_v + (1-λ_p)L_s`` gives a thermodynamically consistent saturation vapor pressure over the mixture.
```

File: docs/src/Formulation.md [hunk 6]
```diff
@@ -0,0 +1,423 @@
...
+
+### Speed of Sound
+The speed of sound in (moist) unstratified air is
+```math
+\begin{equation}
+ c_s = \left(\frac{c_{pm}}{c_{vm}} R_m T \right)^{1/2},
+\label{e:soundspeed}
+\end{equation}
+```
+with the appropriate gas constants for moist air. In the presence of stratification, additional terms arise [Durran1999](@cite).
```
Comment (tapios) [inline docs/src/Formulation.md:423]:
```suggestion
with the appropriate gas constants for moist air. In the presence of stratification, additional terms arise ([Durran1999](@cite)).
```

**Review by tapios — APPROVED** [review]
I like how we can include the conceptual overview in the code. However, now we have the same material in two places: Overleaf design doc and here. If we change one, we should change the other. Is there any way of automatically including LaTeX (e.g., from Overleaf) in the code here, to make it easier to keep them in sync?

**Comment by charleskawczynski** [comment]
> I like how we can include the conceptual overview in the code. However, now we have the same material in two places: Overleaf design doc and here. If we change one, we should change the other. Is there any way of automatically including LaTeX (e.g., from Overleaf) in the code here, to make it easier to keep them in sync?

That's a good point. I agree that having duplicate docs is a bad idea. I don't think there's a way to synchronize them. We could alternatively include a link to the overleaf in the docs, but I don't think the overleaf has public access-- so most users won't be able to view it.

Personally, I'd be fine with using this repo as the primary source for the documentation, and simply referencing this repo's docs in the overleaf-- I feel that the overleaf has more or less converged over the last year. That said, I understand the usefulness of being able to make quick changes in the overleaf. So, unless there's a better alternative, I'm fine with whichever you prefer, @tapios:
 - [ ] Keep these docs, and reference from overleaf
 - [ ] Close this PR, and just stick with the overleaf

**Comment by tapios** [comment]
> > I like how we can include the conceptual overview in the code. However, now we have the same material in two places: Overleaf design doc and here. If we change one, we should change the other. Is there any way of automatically including LaTeX (e.g., from Overleaf) in the code here, to make it easier to keep them in sync?
> 
> That's a good point. I agree that having duplicate docs is a bad idea. I don't think there's a way to synchronize them. We could alternatively include a link to the overleaf in the docs, but I don't think the overleaf has public access-- so most users won't be able to view it.
> 
> Personally, I'd be fine with using this repo as the primary source for the documentation, and simply referencing this repo's docs in the overleaf-- I feel that the overleaf has more or less converged over the last year. That said, I understand the usefulness of being able to make quick changes in the overleaf. So, unless there's a better alternative, I'm fine with whichever you prefer, @tapios:
> 
> * [ ]  Keep these docs, and reference from overleaf
> * [ ]  Close this PR, and just stick with the overleaf

Let's keep the duplicate docs for now. It's not ideal, but acceptable for now. In principle, this documentation can supersede the Overleaf doc. It's just that I think eventually the Overleaf docs can become a CliMA book, which has everything essential on the conceptual side together in one place. So I do not want to give it up yet. Could you just put a note with a link to these docs into the Overleaf, so people know that if they change the Overleaf, they probably should make changes here too.

**Pull request merged**